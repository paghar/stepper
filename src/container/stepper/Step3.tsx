import React, {useEffect,FC} from "react";
import {useForm} from "react-hook-form";
import {addUsersRequest,updateFromStep} from "../../redux/user/userActions";
import {useDispatch} from "react-redux";
import {useRouter} from "next/router";
import {condition} from "../../data/constants";
import {z} from "zod";
import {zodResolver} from "@hookform/resolvers/zod";

const FormSchema = z.object({
    condation: z
        .string({invalid_type_error: "Please accept the condition"})
        .refine((val) => condition.map((item: string) => item).includes(val)),
});  

type FormSchemaType = z.infer<typeof FormSchema>;

const Step2:FC = () => {  
   
    const dispatch = useDispatch(); 
    const router = useRouter();    

    const {
        register,       
        handleSubmit,
        formState: {errors},
    } = useForm<FormSchemaType>({
        resolver: zodResolver(FormSchema),
    });
   
    const onSubmit = (data:any) => {              
        dispatch(addUsersRequest(data));  
        router.push({pathname: "/Stepper", query: {formStep: 4}});       
    };

    useEffect(()=>{
        dispatch(updateFromStep(3));
    },[]);  

    return (
        <form onSubmit={handleSubmit(onSubmit)} className="bg-white shadow-md rounded p-12 m-10">            
            
            <div className="grid grid-cols-2 gap-4">
                <label className="block text-gray-700 text-sm font-bold h-8" >
                    Accept the Condation:
                </label>   
            </div>

            <div>
                <input {...register("condation")} type="radio" value="Yes" />
                <label>Yes</label>                  
            </div>

            <div>
                <input {...register("condation")} type="radio" value="No" />  
                <label>No</label>                  
            </div>

            {errors.condation && (
                <p className="text-sm text-red-600 mt-1">{errors.condation.message}</p>
            )}            
            
            <div className="grid grid-cols-8 mt-4">
                <div className="col-span-7"></div>
                <button 
                    type="submit"
                    className="w-full inline-block px-6 py-2 border-2 border-blue-600 text-blue-600 font-medium
                    text-xs leading-normal uppercase rounded hover:bg-blue hover:bg-opacity-5 focus:outline-none 
                    focus:ring-0 transition duration-150 ease-in-out h-16 "
                >
                 next step
                </button>
            </div>       
           
          
        </form>
    );
};

export default Step2;
