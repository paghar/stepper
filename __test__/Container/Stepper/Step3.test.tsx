import renderer from "react-test-renderer";
import Step3 from "../../../src/container/stepper/Step3";
import {Provider} from "react-redux";

import configureMockStore from "redux-mock-store";

const mockStore = configureMockStore();
const store = mockStore();

jest.mock("next/router", () => ({
    useRouter: jest.fn()
}));

describe(">>>> Stepper", () => {
    test("++++ Snapshot of Stepper3 form", () => {       
        const component = renderer.create(
            <Provider store={store}>
                <Step3/>
            </Provider>
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

export {};

