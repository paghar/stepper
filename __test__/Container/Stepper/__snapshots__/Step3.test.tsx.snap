// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>> Stepper ++++ Snapshot of Stepper1 form 1`] = `
<form
  className="bg-white shadow-md rounded p-12 m-10"
  onSubmit={[Function]}
>
  <div>
    <label
      className="block text-gray-700 text-sm font-bold h-8"
    >
      Accept the Condation:
    </label>
  </div>
  <div>
    <input
      name="condation"
      onBlur={[Function]}
      onChange={[Function]}
      type="radio"
      value="Yes"
    />
    <label>
      Yes
    </label>
  </div>
  <div>
    <input
      name="condation"
      onBlur={[Function]}
      onChange={[Function]}
      type="radio"
      value="No"
    />
    <label>
      No
    </label>
  </div>
  <div
    className="flex flex-row-reverse m-8"
  >
    <button
      className="w-full inline-block px-6 py-2 border-2 border-blue-600 text-blue-600 font-medium
 text-xs leading-normal uppercase rounded hover:bg-blue hover:bg-opacity-5 focus:outline-none 
 focus:ring-0 transition duration-150 ease-in-out h-16 lg:basis-1/6 md:basis-full sm:basis-full basis-full "
      type="submit"
    >
      next step
    </button>
  </div>
</form>
`;

exports[`>>>> Stepper ++++ Snapshot of Stepper3 form 1`] = `
<form
  className="bg-white shadow-md rounded p-12 m-10"
  onSubmit={[Function]}
>
  <div>
    <label
      className="block text-gray-700 text-sm font-bold h-8"
    >
      Accept the Condation:
    </label>
  </div>
  <div>
    <input
      name="condation"
      onBlur={[Function]}
      onChange={[Function]}
      type="radio"
      value="Yes"
    />
    <label>
      Yes
    </label>
  </div>
  <div>
    <input
      name="condation"
      onBlur={[Function]}
      onChange={[Function]}
      type="radio"
      value="No"
    />
    <label>
      No
    </label>
  </div>
  <div
    className="flex flex-row-reverse m-8"
  >
    <button
      className="w-full inline-block px-6 py-2 border-2 border-blue-600 text-blue-600 font-medium
 text-xs leading-normal uppercase rounded hover:bg-blue hover:bg-opacity-5 focus:outline-none 
 focus:ring-0 transition duration-150 ease-in-out h-16 lg:basis-1/6 md:basis-full sm:basis-full basis-full "
      type="submit"
    >
      next step
    </button>
  </div>
</form>
`;
